### PR TITLE
Feature/spatialized sound

### PR DESCRIPTION
**Pull request by Korotsu**

### CODE REVIEW

File: editor/src/EditorStartup.cpp [hunk 1]
```diff
@@ -46,29 +46,26 @@ void EditorStartup::initializeDefaultInputs() const
...
 EditorStartup::EditorStartup()
-    : m_fixedUpdate{[&](double fixedUnscaledDeltaTime, double fixedDeltaTime)
-      {
+    : m_fixedUpdate{[&](double fixedUnscaledDeltaTime, double fixedDeltaTime) {
           m_engine->physXSystem.advance(fixedDeltaTime);
       }},
       m_update{[&](double unscaledDeltaTime, double deltaTime) {
           m_engine->sceneManager.update();
           m_engine->inputManager.processInput();
+          m_engine->soundSystem.update();
```
Comment (badsami) [inline editor/src/EditorStartup.cpp:55]: There never is a situation when we can play sound in the editor without simulating the game being played.
I believe it is unnecessary to update the sound system here.

```suggestion
```

File: editor/src/EditorStartup.cpp [hunk 2]
```diff
@@ -245,6 +243,7 @@ void EditorStartup::pauseGame()
             auto updateSceneManagerFunct = GET_PROCESS(m_reloadableCpp, updateSceneManager);
             updateSceneManagerFunct();
             m_engine->inputManager.processInput();
+            m_engine->soundSystem.update();
```
Comment (badsami) [inline editor/src/EditorStartup.cpp:246]: There never is a situation when we can play sound in the editor while the game simulation is paused.
I believe it is unnecessary to update the sound system here.

```suggestion
```

File: editor/src/EditorStartup.cpp [hunk 3]
```diff
@@ -273,6 +272,7 @@ void EditorStartup::stopGame()
             updateSceneManagerFunct();
 
             m_engine->inputManager.processInput();
+            m_engine->soundSystem.update();
```
Comment (badsami) [inline editor/src/EditorStartup.cpp:275]: There never is a situation when we can play sound in the editor without simulating the game.
I believe it is unnecessary to update the sound system here.

```suggestion
```

**Review by badsami — CHANGES_REQUESTED** [review]
A few bugs:
- Changing the pitch of a sound and other parameters unrelated to sound spatialization seem to change the position and/or spatialization of the sound
- `m_engine->soundSystem.update()` should only be called when playing the game (see comments on the code of these commits)

File: editor/src/EditorStartup.cpp (hunk 1 above)
Comment (Korotsu) [reply editor/src/EditorStartup.cpp:55]: It's not about playing sound but updating settings within the editor.

File: editor/src/EditorStartup.cpp (hunk 2 above)
Comment (Korotsu) [reply editor/src/EditorStartup.cpp:246]: same than before

File: editor/src/EditorStartup.cpp (hunk 3 above)
Comment (Korotsu) [reply editor/src/EditorStartup.cpp:275]: same than before

**Review by badsami — APPROVED** [review]
After discussing the problems at hand, we clarified a few points and came to the conclusion that nothing can be done on our side about the pitch and other parameters affecting sound spatialization.

We also clarified what soundSystem.update() does.

Approved.

**Pull request merged**